.github/workflows/promote_to_branch.yml

```diff
@@ -0,0 +1,78 @@
+name: Run Test on push to branch
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!master'
+
+jobs:
+
+  # This action (dorny/paths-filter) allows us to selectively run jobs based on
+  # the files that have been updated, as in on.push.paths above.
+  # Saves us from doing unnecessary builds.
+  # ---------------------------------------------------------------------------
+  
+  api_test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./api
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12"
+      - name: Install dependencies
+        run: npm ci
+      - name: Unit tests w/ coverage
+        run: npm run test:coverage
+      - name: LINTing
+        run: npm run test:lint
+      - name: OpenAPI Schema
+        run: npm run test:schema
+      - name: Upload coverage report
+        env:
+          CC_TEST_REPORTER_ID: ${{ secrets.TestReporterID }}
+          CI: "true"
+        run: |
+          curl -Ls https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+          chmod +x ./cc-test-reporter
+          ./cc-test-reporter format-coverage -t lcov -o codeclimate.json
+          ./cc-test-reporter upload-coverage -i codeclimate.json
+
+      - name: Check test result
+        if: ${{ failure() }}
+        env:
+          ROCKETCHAT_MESSAGE: "There was an error in the registry app API tests during the ${{ github.workflow }} workflow.  See: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+        run: |
+          echo '{"text":"${{ env.ROCKETCHAT_MESSAGE }}"}' "${{ secrets.ROCKETCHAT_HOOK_URL }}"
+          exit 1
+  web_test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./web
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12"
+      - name: Install dependencies
+        run: npm ci
+      - name: test
+        run: ls src/__tests__/__snapshots__/
```

Comment:
I am unsure of the value this step adds. Any particular reason for its inclusion?